Exceptions with 1.17.1

<!--
Please make sure to first check our common issues page before creating a new issue:
https://github.com/Imprex-Development/Orebfuscator/wiki/Common-Issues
-->

**Please fill out the following information:**

* Server Version: git-Paper-408 (MC: 1.17.1)
* Orebfuscator Version: 5.2.3
* Protocollib Version: 4.7.0

**Describe the bug**

I get multiple of the following exceptions. They appear to happen when teleporting. orebfuscator_cache was cleared before starting the server.
```
2022-01-06 10:12:22 [WARN] java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.level.chunk.ChunkSection.c()" because "section" is null
2022-01-06 10:12:22 [WARN] 	at java.base/java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:367)
2022-01-06 10:12:22 [WARN] 	at java.base/java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:376)
2022-01-06 10:12:22 [WARN] 	at java.base/java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:1093)
2022-01-06 10:12:22 [WARN] 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
2022-01-06 10:12:22 [WARN] 	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2158)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
2022-01-06 10:12:22 [WARN] 	at java.base/java.lang.Thread.run(Thread.java:831)
2022-01-06 10:12:22 [WARN] Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.level.chunk.ChunkSection.c()" because "section" is null
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.nms.v1_17_R1.ReadOnlyChunkWrapper.getState(ReadOnlyChunkWrapper.java:22)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.nms.v1_17_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:31)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:122)
2022-01-06 10:12:22 [WARN] 	at Orebfuscator.jar//net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
2022-01-06 10:12:22 [WARN] 	... 2 more
```

--- a/orebfuscator-nms/orebfuscator-nms-v1_17_R1/src/main/java/net/imprex/orebfuscator/nms/v1_17_R1/ReadOnlyChunkWrapper.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_17_R1/src/main/java/net/imprex/orebfuscator/nms/v1_17_R1/ReadOnlyChunkWrapper.java
@@ -19,7 +19,7 @@ public class ReadOnlyChunkWrapper implements ReadOnlyChunk {
 		LevelChunkSection[] sections = chunk.getSections();
 		if (sectionIndex >= 0 && sectionIndex < sections.length) {
 			LevelChunkSection section = sections[sectionIndex];
-			if (!section.isEmpty()) {
+			if (!LevelChunkSection.isEmpty(section)) {
 				return section.getBlockState(x & 0xF, y & 0xF, z & 0xF);
 			}
 		}

--- a/orebfuscator-nms/orebfuscator-nms-v1_18_R1/src/main/java/net/imprex/orebfuscator/nms/v1_18_R1/ReadOnlyChunkWrapper.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_18_R1/src/main/java/net/imprex/orebfuscator/nms/v1_18_R1/ReadOnlyChunkWrapper.java
@@ -19,7 +19,7 @@ public class ReadOnlyChunkWrapper implements ReadOnlyChunk {
 		LevelChunkSection[] sections = chunk.getSections();
 		if (sectionIndex >= 0 && sectionIndex < sections.length) {
 			LevelChunkSection section = sections[sectionIndex];
-			if (!section.hasOnlyAir()) {
+			if (sections != null && !section.hasOnlyAir()) {
 				return section.getBlockState(x & 0xF, y & 0xF, z & 0xF);
 			}
 		}

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/chunk/Chunk.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/chunk/Chunk.java
@@ -55,7 +55,7 @@ public class Chunk implements AutoCloseable {
 		return null;
 	}
 
-	public int getBlock(int x, int y, int z) {
+	public int getBlockState(int x, int y, int z) {
 		if (x >> 4 == this.chunkX && z >> 4 == this.chunkZ) {
 			ChunkSectionHolder chunkSection = this.sections[this.heightAccessor.getSectionIndex(y)];
 			if (chunkSection != null) {

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/chunk/ChunkSection.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/chunk/ChunkSection.java
@@ -59,12 +59,12 @@ public class ChunkSection {
 		return y << 8 | z << 4 | x;
 	}
 
-	public void setBlock(int x, int y, int z, int blockId) {
-		this.setBlock(positionToIndex(x, y, z), blockId);
+	public void setBlockState(int x, int y, int z, int blockId) {
+		this.setBlockState(positionToIndex(x, y, z), blockId);
 	}
 
-	public void setBlock(int index, int blockId) {
-		int prevBlockId = this.getBlock(index);
+	public void setBlockState(int index, int blockId) {
+		int prevBlockId = this.getBlockState(index);
 
 		if (!NmsInstance.isAir(prevBlockId)) {
 			--this.blockCount;
@@ -79,10 +79,10 @@ public class ChunkSection {
 	}
 
 	public int getBlock(int x, int y, int z) {
-		return this.getBlock(positionToIndex(x, y, z));
+		return this.getBlockState(positionToIndex(x, y, z));
 	}
 
-	public int getBlock(int index) {
+	public int getBlockState(int index) {
 		return this.palette.valueFor(this.data.get(index));
 	}
 
@@ -126,7 +126,7 @@ public class ChunkSection {
 
 		int[] directData = new int[4096];
 		for (int i = 0; i < directData.length; i++) {
-			directData[i] = this.getBlock(i);
+			directData[i] = this.getBlockState(i);
 		}
 		return directData;
 	}

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationProcessor.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationProcessor.java
@@ -58,7 +58,7 @@ public class ObfuscationProcessor {
 						continue;
 					}
 
-					int blockState = chunkSection.getBlock(index);
+					int blockState = chunkSection.getBlockState(index);
 
 					int obfuscateBits = blockFlags.flags(blockState, y);
 					if (BlockFlags.isEmpty(obfuscateBits)) {
@@ -81,7 +81,7 @@ public class ObfuscationProcessor {
 					if (!obfuscated && BlockFlags.isProximityBitSet(obfuscateBits) && proximityConfig.shouldObfuscate(y)) {
 						proximityBlocks.add(new BlockPos(x, y, z));
 						if (BlockFlags.isUseBlockBelowBitSet(obfuscateBits)) {
-							blockState = getBlockBelow(blockFlags, chunk, x, y, z);
+							blockState = getBlockStateBelow(blockFlags, chunk, x, y, z);
 						} else {
 							blockState = proximityConfig.nextRandomBlockState();
 						}
@@ -90,7 +90,7 @@ public class ObfuscationProcessor {
 
 					// update block state if needed
 					if (obfuscated) {
-						chunkSection.setBlock(index, blockState);
+						chunkSection.setBlockState(index, blockState);
 						if (BlockFlags.isBlockEntityBitSet(obfuscateBits)) {
 							blockEntities.add(new BlockPos(x, y, z));
 						}
@@ -106,9 +106,9 @@ public class ObfuscationProcessor {
 
 	// returns first block below given position that wouldn't be obfuscated in any
 	// way at given position
-	private int getBlockBelow(BlockFlags blockFlags, Chunk chunk, int x, int y, int z) {
+	private int getBlockStateBelow(BlockFlags blockFlags, Chunk chunk, int x, int y, int z) {
 		for (int targetY = y - 1; targetY > chunk.getHeightAccessor().getMinBuildHeight(); targetY--) {
-			int blockData = chunk.getBlock(x, targetY, z);
+			int blockData = chunk.getBlockState(x, targetY, z);
 			if (blockData != -1 && BlockFlags.isEmpty(blockFlags.flags(blockData, y))) {
 				return blockData;
 			}
@@ -130,7 +130,7 @@ public class ObfuscationProcessor {
 			return false;
 		}
 
-		int blockId = chunk.getBlock(x, y, z);
+		int blockId = chunk.getBlockState(x, y, z);
 		if (blockId == -1) {
 			blockId = task.getBlockState(x, y, z);
 		}

